Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt

```diff
@@ -229,25 +230,119 @@ internal class EnumStubBuilder(
                 origin = origin,
-                interfaces = listOf(context.platform.getRuntimeType("CEnum"))
+                interfaces = listOf(context.platform.getRuntimeType("CEnum")),
+                childrenClasses = listOfNotNull(enumVarClass)
         )
         context.bridgeComponentsBuilder.enumToTypeMirror[enum] = baseTypeMirror
-
         return listOf(enum)
     }
 
+    private fun constructAliasProperty(enumConstant: EnumConstant, entry: EnumEntryStub): PropertyStub {
+        val aliasAnnotation = AnnotationStub.CEnumEntryAlias(entry.name)
+                .takeIf { context.generationMode == GenerationMode.METADATA }
+        return PropertyStub(
+                enumConstant.name,
+                ClassifierStubType(classifier),
+                kind = PropertyStub.Kind.Val(PropertyAccessor.Getter.GetEnumEntry(entry)),
+                origin = StubOrigin.EnumEntry(enumConstant),
+                annotations = listOfNotNull(aliasAnnotation)
+        )
+    }
+
+    private fun constructEnumVarClass(): ClassStub.Simple {
+
+        val enumVarClassifier = classifier.nested("Var")
+
+        val rawPtrConstructorParam = FunctionParameterStub("rawPtr", context.platform.getRuntimeType("NativePtr"))
+        val superClass = context.platform.getRuntimeType("CEnumVar")
+        require(superClass is ClassifierStubType)
+        val primaryConstructor = ConstructorStub(
+                parameters = listOf(rawPtrConstructorParam),
+                isPrimary = true,
+                annotations = emptyList(),
+                origin = StubOrigin.Synthetic.DefaultConstructor
+        )
+        val superClassInit = SuperClassInit(superClass, listOf(GetConstructorParameter(rawPtrConstructorParam)))
+
+        val baseIntegerTypeSize = when (val unwrappedType = enumDef.baseType.unwrapTypedefs()) {
+            is IntegerType -> unwrappedType.size.toLong()
+            CharType -> 1L
+            else -> error("Incorrect base type for enum ${classifier.fqName}")
+        }
+        val typeSize = IntegralConstantStub(baseIntegerTypeSize, 4, true)
+        val companionSuper = (context.platform.getRuntimeType("CPrimitiveVar") as ClassifierStubType).nested("Type")
+        val varSizeAnnotation = AnnotationStub.CEnumVarTypeSize(baseIntegerTypeSize.toInt())
+                .takeIf { context.generationMode == GenerationMode.METADATA }
+        val companion = ClassStub.Companion(
+                classifier = enumVarClassifier.nested("Companion"),
+                superClassInit = SuperClassInit(companionSuper, listOf(typeSize)),
+                annotations = listOfNotNull(varSizeAnnotation)
+        )
+        val valueProperty = PropertyStub(
+                name = "value",
+                type = ClassifierStubType(classifier),
+                kind = PropertyStub.Kind.Var(
+                        PropertyAccessor.Getter.ExternalGetter(),
+                        PropertyAccessor.Setter.ExternalSetter()
```

Comment:
Are these accessors really external?

Reply:
Yes. Similar to `@CCall` functions, accessors are generated at callsite. See https://github.com/JetBrains/kotlin-native/pull/3800/commits/93ee3d33f154253993deb34097d62096b465aa14#diff-fd228a115334072fe2f9ad51d9b8d9b9R163